singularity-eos/eos/get_sg_eos_lambdas.hpp

```diff
@@ -0,0 +1,136 @@
+//------------------------------------------------------------------------------
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_GET_SG_EOS_LAMBDAS_HPP_
+#define _SINGULARITY_EOS_EOS_GET_SG_EOS_LAMBDAS_HPP_
+
+#include <ports-of-call/portability.hpp>
+
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+
+#define SG_GET_SG_EOS_INIT_LAMBDA_DECL                                                   \
```

Comment:
So I'm willing to let this go in, but I find macros declaring million-line lambdas, pretty distasteful. Is there some way this could be cleaned up and made easier to modify the next time someone needs to change this code? Maybe declare inner loop functions, or turn these into functors, or something?